android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/RpcClient.java

```diff
@@ -316,12 +317,12 @@ protected void sendRequest(String request) throws IOException {
             Log.d(Logging.TAG, "mRequest.capacity() = " + mRequest.capacity());
         if (Logging.RPC_DATA && Logging.DEBUG)
             Log.d(Logging.TAG, "Sending request: \n" + request);
-        if (mOutput == null)
+        if (socketSink == null)
             return;
-        mOutput.write("<boinc_gui_rpc_request>\n");
-        mOutput.write(request);
-        mOutput.write("</boinc_gui_rpc_request>\n\003");
-        mOutput.flush();
+        socketSink.writeString("<boinc_gui_rpc_request>\n", Charsets.ISO_8859_1);
```

Comment:
Is there a way to set global or default charset in Okio and not set it on every write?

Reply:
Why there was `'\003'` char? Is needed now?

Reply:
It's still here (two lines above)

Reply:
Why we need this char?

Reply:
> 
> 
> Is there a way to set global or default charset in Okio and not set it on every write?

Okio uses UTF-8 by default.

Reply:
You can do it like this:
https://javarevisited.blogspot.com/2012/01/get-set-default-character-encoding.html

Reply:
> 
> 
> You can do it like this:
> https://javarevisited.blogspot.com/2012/01/get-set-default-character-encoding.html

Thanks for the suggestion, but I don't think such a change is necessary. The charset only needs to be specified for 3 lines of code.